extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java

```diff
@@ -94,6 +96,14 @@
     private static final String DISABLE_SMART_PRODUCES_QUARKUS = "quarkus.rest-client.disable-smart-produces";
     private static final String KOTLIN_INTERFACE_DEFAULT_IMPL_SUFFIX = "$DefaultImpls";
 
+    private static final Set<DotName> SKIP_COPYING_ANNOTATIONS_TO_GENERATED_CLASS = Set.of(
+            REGISTER_REST_CLIENT,
+            REGISTER_PROVIDER,
+            REGISTER_PROVIDERS,
+            CLIENT_HEADER_PARAM,
+            CLIENT_HEADER_PARAMS,
+            REGISTER_CLIENT_HEADERS);
```

Comment:
Hi @michalszynkiewicz, I could use your help here :-)

Basically what I do in this PR is that I take _all_ annotations from the RestClient interface and copy them to the generated class, with the exception of several well-known RestClient annotations defined here. Does that list make sense to you?

(Alternatively, I could only copy interceptor binding annotations. That would fix the original bug as well, but I have this hunch that some other cases would be left uncovered. Technically, thinking about it, in the present state, one could even put a scope annotation on the interface and it would be propagated to the generated class, which I'm not sure is expected, but it feels kinda nice :-) )

Reply:
(Actually looking at it, if someone added a scope annotation to the interface, it would indeed be copied to the generated class, where it would clash with another scope annotation generated by this processor. So that's a terrible example and perhaps a reason to reconsider...)

Reply:
I think it would be nice to support scope annotations on interfaces, the interceptor can be modified to not add in this case, no?

Reply:
Scope annotations are already supported, the `computeDefaultScope` method looks for them on the interface (unless there's a configuration override).